util/util.go

```diff
@@ -58,3 +63,50 @@ func AddImageNames(store storage.Store, image *storage.Image, addNames []string)
 	}
 	return nil
 }
+
+func ContextFromGlobalOptions(c *cli.Context, flagPrefix string) (*types.SystemContext, error) {
+	ctx := &types.SystemContext{
+		RegistriesDirPath: c.GlobalString("registries.d"),
+		DockerCertPath:    c.String(flagPrefix + "cert-dir"),
+		// DEPRECATED: keep this here for backward compatibility, but override
+		// them if per subcommand flags are provided (see below).
+		DockerInsecureSkipTLSVerify: !c.GlobalBoolT("tls-verify"),
+		OSTreeTmpDirPath:            c.String(flagPrefix + "ostree-tmp-dir"),
+	}
+	if c.IsSet(flagPrefix + "tls-verify") {
+		ctx.DockerInsecureSkipTLSVerify = !c.BoolT(flagPrefix + "tls-verify")
```

Comment:
I'd not considered that for this submit.  I was just focusing on getting the un/pwd bit working.  So in total should I be trying to get the tls-verify, registries.d, cert-dir, and os-temp-tree all working as new options?  Would it be ok to just get un/pwd working for this, then also for commit/push, then circle back and do the tls, registries.d, cert-dir and os-temp-tree as another commit?  Do you think we need them all?